lib/schnorr.py

```diff
@@ -118,7 +118,7 @@ def nonce_function_rfc6979(order, privkeybytes, msg32, algo16=b'', ndata=b''):
     return k
 
 
-def sign(privkey, message_hash):
+def sign(privkey, message_hash,ndata=b''):
```

Comment:
`ndata` needs to default to `None`.  If it is not `None`, it needs to be exactly 32 bytes, otherwise an exception should be thrown.  This invariant needs to be enforced all over the place where you added this `ndata` argument.